src/portable/portable.c

```diff
@@ -706,6 +707,14 @@ static int install_chroot_dropin(
 
                         return -ENOMEM;
 
+                r = load_os_release_pairs_with_prefix("/", "service_host_", &os_release_pairs);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to read os-release from host for portable service, ignoring: %m");
+                STRV_FOREACH(s, os_release_pairs)
+                        if (!strextend(&text, "Environment=", *s, "\n", NULL))
+                                return -ENOMEM;
+        }
```

Comment:
as mentioned, I think we probably should do the /run/host thing only for portable services.

to summarize:

1. in nspawn implement both `$container_id`, `$container_version_id`, … *and* /run/host
2. in portable services implement only /run/host

Does that make sense?

Reply:
Ok, makes sense, thanks - I'll work on it and push again